### PR TITLE
Feature8/integration

### PR DESCRIPTION
**Pull request by techEmprez**
### **Implemented the following in this milestone**

-  Solved the N+1 problem when fetching all posts and their comments for a user.

-  Used Capybara to write integration tests for each view in your project which includes ;

- [ ] User index page
- [ ] User show page
- [ ] Post index page
- [ ] Post show page

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +51,8 @@ gem 'rubocop', '>= 1.0', '< 2.0'
 
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
+  gem 'bullet'
+  gem 'database_cleaner'
   gem 'debug', platforms: %i[mri mingw x64_mingw]
 end
```
Comment (Owusu-Desmond) [inline Gemfile]: - [ ] Please make sure that you have the `Capybara` gem in `Gemfile` for the integration test :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @techEmprez and @quavo19 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please make sure that all your tests are passing :pray: 
  Screenshot from 2023-05-16 23-55-44


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (techEmprez) [reply Gemfile]: All done...
All test cases passed

Comment (techEmprez) [reply Gemfile]: `gem 'capybara'` exist

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @techEmprez and @quavo19, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to fix the failing tests.

## Optional suggestions

- [ ] Let's always remember that in a Rails app, we also need CSS linters. I have noticed that you only have a Rubocop linter, if this happens in the next milestones, your PR will be invalidated.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**